test/utils/catalystList.spec.ts

```diff
@@ -0,0 +1,101 @@
+import { getUpdatedApprovedListWithoutQueryingContract } from '../../src/utils/catalystList'
+
+describe('Catalyst Client Builder', () => {
+  describe('getUpdatedApprovedListWithoutQueryingContract', () => {
+    const ORIGIN = 'origin'
+    const REQUIRED_LISTS = 2
+    const [ADDRESS1, ADDRESS2, ADDRESS3] = ['http://test1.com', 'http://test2.com', 'http://test3.com']
+
+    it('When the amount of known servers on the list is less than required, then no fetching is performed and nothing is returned', async () => {
+      let fetchCount = 0
+
+      const result = await calculateList({
+        list: [{ address: ADDRESS1 }],
+        fetchApprovedCatalysts: () => {
+          fetchCount++
+          return Promise.resolve([])
+        }
+      })
+
+      expect(fetchCount).toEqual(0)
+      expect(result).toBeUndefined()
+    })
+
+    it(`When one of the first N servers doesn't respond, then next one is queried`, async () => {
+      let fetchCount = 0
+
+      await calculateList({
+        list: [{ address: ADDRESS1 }, { address: ADDRESS2 }, { address: ADDRESS3 }],
+        fetchApprovedCatalysts: () => {
+          if (fetchCount++ === 0) {
+            return Promise.resolve(undefined)
+          }
+          return Promise.resolve([ADDRESS1])
+        }
+      })
+
+      expect(fetchCount).toEqual(3)
```

Comment:
If only the first one failed, and the second one retrieved an address then why got the third one called too?